apollo-api/build.gradle.kts

```diff
@@ -36,6 +44,7 @@ kotlin {
       dependsOn(commonMain)
       dependencies {
         implementation(kotlin("stdlib"))
+        implementation(groovy.util.Eval.x(project, "x.dep.kotlin.reflect"))
```

Comment:
I'm super reluctant to adding `kotlin-reflect` as a dependency as it's known to be a laaaaaarge dependency. See https://youtrack.jetbrains.com/issue/KT-20793 for an exemple.

`apollo-api` is pretty lightweight at the moment and given that it's imported by all other modules, I'd like to keep it that way. 

Can you elaborate why keeping a String for scalar types doesn't work ? I didn't find that many places using `KClass.qualifiedName()` and most of them are for debug/troubleshooting.

Reply:
[The dependency shouldn't be required since 1.3.70](https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/), since KClass is moved out of `kotlin-reflect` jar. I will remove it.

> Can you elaborate why keeping a String for scalar types doesn't work ? I didn't find that many places using KClass.qualifiedName() and most of them are for debug/troubleshooting.

It is used by `apollo-api` to select the type adapters to use (among the set of default ones for Kotlin/Java stdlib types, and the user-supplied ones via `customTypeMapping` in Gradle).

Reply:
>The dependency shouldn't be required since 1.3.70, since KClass is moved out of kotlin-reflect jar. I will remove it.

Nice 👍 

> The lookup tables are currently keyed by KClass.qualifiedName(), which does not work at Kotlin/JS runtime.

That's what I'm trying to understand. 

Looking at `ScalarTypeAdapters`:

```kotlin
  fun <T : Any> adapterFor(scalarType: ScalarType): CustomTypeAdapter<T> {
    var customTypeAdapter: CustomTypeAdapter<*>? = customTypeAdapters[scalarType.typeName()]
    if (customTypeAdapter == null) {
      customTypeAdapter = DEFAULT_ADAPTERS[scalarType.className()]
    }
    return requireNotNull(customTypeAdapter) {
      "Can't map GraphQL type: `${scalarType.typeName()}` to: `${scalarType.className()}`. Did you forget to add a custom type adapter?"
    } as CustomTypeAdapter<T>
  }
```

This doesn't use `KClass.qualifiedName()`, does it ? both `typeName()` and `className()`are generated by the compiler base on either default hardcoded value (`java.lang.String`, etc...) or user input in case of a custom adapter:

```kotlin
enum class CustomType : ScalarType {
  ID {
    override fun typeName(): String = "ID"

    override fun className(): String = "kotlin.String"
  },

  URL {
    override fun typeName(): String = "URL"

    override fun className(): String = "kotlin.Any"
  }
}
```

Reply:
That's a good point. I probably had made the decision to change it simply because Gradle wasn't building, ignoring the fact that `.qualifiedName()` is used only in Tests. I should revert it.